packages/frontend/components/primitives/menu-dropdown/component.tsx

```diff
@@ -0,0 +1,46 @@
+import { useOutsideClick } from '@hooks';
+import { useState } from 'react';
+import * as styles from './styles';
+import type { DropdownProps } from './types';
+
+export const Dropdown = ({ options, ...props }: DropdownProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  const ref = useOutsideClick(() => setIsOpen(false));
```

Comment:
According to your implementation of useOutsideClick it make sense to wrap callback () => setIsOpen(false) in useCallback:

const handleClickOutside = useCallback(() => setIsOpen(false))